compiler.cpp

```diff
@@ -408,49 +365,51 @@ std::ostream& operator<<(std::ostream& out, ParseState& PS) {
   return out;
 }
 
-#define EAT(PS, t) if (PS.eat() != (t)) { return LogError("EAT ERROR"); }
+#define EAT(PS, t)                                                             \
```

Comment:
I know this is a remnant, but maybe we should consider not using macros to save the students